### PR TITLE
Bump version

### PR DESCRIPTION
**Pull request by BradleyBoutcher**
### What does this PR do?
- Updated Go Dependencies
- Updated Ruby Gems and dependencies
- Modified import statement in `world.rb` to meet newer dependency requirements
- Modified string building in `test.sh` to work with newer versions of cucumber

### What ticket does this PR close?
Connected to #66 

#### Change log
- [ ] The CHANGELOG has been updated, or
- [X] This PR does not include user-facing changes and doesn't require a CHANGELOG update

#### Test coverage
- [ ] This PR includes new unit and integration tests to go with the code changes, or
- [X] The changes in this PR do not require tests

#### Documentation
- [ ] Docs (e.g. `README`s) were updated in this PR, and/or there is a follow-on issue to update docs, or
- [X] This PR does not require updating any documentation

### CODE REVIEW

File: test.sh [hunk 1]
```diff
@@ -18,7 +18,10 @@ export BRANCH_NAME=${BRANCH_NAME:-$(git symbolic-ref --short HEAD)}
 
 # Skip the integration tests if the Summon variables are not present
 if [ -z "$CF_API_ENDPOINT" ]; then
-    INTEGRATION_TAG="--tags ~@integration"
+    # Due to single quote requirements for multi-word tag arguments,
+    # this formatting is necessary
+    ARG="--tags"
+    INT="not @integration"
```
Comment (izgeri) [inline test.sh:24]: should these be set to empty vars if the code doesn't hit this path?

File: conjur-env/go.mod [hunk 2]
```diff
@@ -2,10 +2,12 @@ module github.com/cyberark/cloudfoundry-conjur-buildpack/conjur-env
 
 require (
 	github.com/cyberark/conjur-api-go v0.6.0
-	github.com/cyberark/summon v0.8.0
-	github.com/sirupsen/logrus v1.4.2 // indirect
-	golang.org/x/sys v0.0.0-20200124204421-9fbb57f87de9 // indirect
-	gopkg.in/yaml.v2 v2.2.8 // indirect
+	github.com/cyberark/summon v0.8.2
```
Comment (izgeri) [inline conjur-env/go.mod:5]: fantastic! can you please also update the NOTICES in this PR? it will make it easier for when we release

I _think_ based on my more recent understanding of modules, that we can exclude the `indirect` deps from the NOTICES (since we document first-tier, manually included deps only)

when you make the updates, please verify that the copyright snippets are still valid for each dep.

File: test.sh (hunk 1 above)
Comment (BradleyBoutcher) [reply test.sh]: The shell will disregard them if they aren't set, which they will not be if we don't hit this path.
In any case, there were more issues with string interpolation, so I modified the command.

File: conjur-env/go.mod (hunk 2 above)
Comment (BradleyBoutcher) [reply conjur-env/go.mod:5]: It looks like the only change was related to the summon version, so I added that to the PR 👍

File: test.sh [hunk 3]
```diff
@@ -35,11 +33,21 @@ docker-compose build
...
- --out ./features/reports \
- ${INTEGRATION_TAG}
+if [ -n "$CF_API_ENDPOINT" ]; then
+     # run tests against latest build of buildpack
+     docker-compose run --rm \
+     -w "$BUILDPACK_ROOT_DIR/ci" \
+     tester cucumber \
+     --format pretty \
+     --format junit \
+     --out ./features/reports
```
Comment (izgeri) [inline test.sh:43]: I think these lines should be more indented

File: test.sh [hunk 4]
```diff
@@ -35,11 +33,21 @@ docker-compose build
...
+else
+    # Skip the integration tests if the Summon variables are not present  docker-compose run --rm
+    docker-compose run --rm \
+    -w "$BUILDPACK_ROOT_DIR/ci" \
+    tester cucumber \
+    --format pretty \
+    --format junit \
+    --out ./features/reports \
+    --tags 'not @integration'
```
Comment (izgeri) [inline test.sh:52]: same, this needs more indentation

File: test.sh [hunk 5]
```diff
@@ -35,11 +33,21 @@ docker-compose build
...
+if [ -n "$CF_API_ENDPOINT" ]; then
+     # run tests against latest build of buildpack
+     docker-compose run --rm \
+     -w "$BUILDPACK_ROOT_DIR/ci" \
+     tester cucumber \
+     --format pretty \
+     --format junit \
+     --out ./features/reports
+else
+    # Skip the integration tests if the Summon variables are not present  docker-compose run --rm
```
Comment (izgeri) [inline test.sh:45]:
```suggestion
    # Skip the integration tests if the Summon variables are not present
```

File: test.sh [hunk 6]
```diff
@@ -35,11 +33,21 @@ docker-compose build
...
-# run tests against latest build of buildpack
-docker-compose run --rm \
- -w "$BUILDPACK_ROOT_DIR/ci" \
- tester cucumber \
- --format pretty \
- --format junit \
- --out ./features/reports \
- ${INTEGRATION_TAG}
+if [ -n "$CF_API_ENDPOINT" ]; then
+     # run tests against latest build of buildpack
```
Comment (izgeri) [inline test.sh:37]: this is 100% an ignorable nit but now you've capitalized one comment and I want it for all the others too
```suggestion
     # Run tests against latest build of buildpack (including integration tests against remote foundation)
```

File: NOTICES.txt [hunk 7]
```diff
@@ -120,7 +120,8 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 MIT License is applicable to the following component(s).
 
 
->>> github.com/cyberark/summon-0.8.0
+>>> github.com/cyberark/summon-0.8.2
```
Comment (izgeri) [inline NOTICES.txt:123]:
```suggestion
```
line was doubled

File: test.sh [hunk 8]
```diff
@@ -29,17 +27,27 @@ else
...
+         --out ./features/reports
+else
+    # Skip the integration tests if the Summon variables are not present
+    docker-compose run --rm \
+        -w "$BUILDPACK_ROOT_DIR/ci" \
+        tester cucumber \
+        --format pretty \
+        --format junit \
+        --out ./features/reports \
+        --tags 'not @integration'
```
Comment (izgeri) [inline test.sh:52]: why did you change the syntax from `~@integration`?

File: test.sh (hunk 6 above)
Comment (izgeri) [reply test.sh]: @BradleyBoutcher I still think it'd be helpful to add something like I suggested to the end of this line
> (including integration tests against remote foundation)

to clarify that in this flow, we are also running those tests

File: test.sh (hunk 8 above)
Comment (BradleyBoutcher) [reply test.sh:52]: That is the syntax required for newer versions of cucumber. Additionally, if you use multi-word tag statements, such as in this case, cucumber requires it to be inside single quotes. There's no easy way to store single quotes inside a string and then pass them into this docker-compose statement, and this is the only solution I could find after spending a good deal of time on it.

File: test.sh (hunk 6 above)
Comment (BradleyBoutcher) [reply test.sh]: Ah thank you! I didn't see the rest of this change and only capitalized the letter. I'll commit this change 👍
